### PR TITLE
flex: Add recipe to build flex

### PR DESCRIPTION
**Pull request by jakirkham**
Builds flex for Linux and Mac. Borrowed with some modifications from conda recipes.

### CODE REVIEW

File: recipes/flex/build.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+./configure --prefix="$PREFIX"
+make
+make install
```
Comment (ocefpaf) [inline recipes/flex/build.sh]: `make check` :stuck_out_tongue_closed_eyes:

File: recipes/flex/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  name: flex
+  version: {{ version }}
+
+source:
+  fn: flex-{{ version }}.tar.gz
+  url: http://sourceforge.net/projects/flex/files/flex-{{ version }}.tar.gz
+  md5: e133e9ead8ec0a58d81166b461244fde
+
+build:
+  number: 1
```
Comment (ocefpaf) [inline recipes/flex/meta.yaml]: 0 :stuck_out_tongue_closed_eyes:

Comment (jakirkham) [inline recipes/flex/meta.yaml]: Oops. Yeah, I got most things, but did miss that one. Good eye.

File: recipes/flex/build.sh (hunk 1 above)
Comment (jakirkham) [inline recipes/flex/build.sh]: Yeah, I had pushed, but GitHub is being flaky with this PR.

**Comment by jakirkham** [comment]
Weird. Travis CI is not being run. Circle CI can't find the commit. AppVeyor is doing something when it shouldn't.

**Comment by ocefpaf** [comment]
> Weird. Travis CI is not being run. Circle CI can't find the commit. AppVeyor is doing something when it shouldn't.

I guess we are abusing the CIs today. Let's give the :construction_worker: a break. I will re-start these jobs soon.

**Comment by jakirkham** [comment]
I don't know if the CI's are at fault for this one. The PR got created with an old commit when I had pushed a new one. So, am kind of thing this PR is broken. It also has been weird with every commit I have pushed since. Slow to notice the new commit, unwilling to force push and such. Let me put this in a new PR and close this one out. If it still doesn't work, we can wait.

**Comment by jakirkham** [comment]
Added here ( https://github.com/conda-forge/staged-recipes/pull/121 ).

**Pull request closed without merging**